store/storage/store.go

```diff
@@ -61,7 +72,17 @@ func (ss *StorageStore) ApplyChangeset(version uint64, cs *store.Changeset) erro
 		}
 	}
 
-	return b.Write()
+	if err := b.Write(); err != nil {
+		return err
+	}
+
+	if prune, pruneVersion := ss.pruneOptions.ShouldPrune(version); prune {
```

Comment:
been off for a week what was the outcome here? async vs sync is handled by the underlying db implementation right?

Reply:
yes